test/models/element_test.rb

```diff
@@ -97,21 +99,32 @@ def setup
     top_form = elements(:one)
     top_form.label = 'form'
     top_form.save!
-    Element.create(workflow_id: workflow.id, label: 'input_text',
-                   element_type_id: 2, parent_id: top_form.id)
-    fs1 = Element.create(workflow_id: workflow.id, label: 'fieldset_one',
-                         element_type_id: 7, parent_id: top_form.id)
-    Element.create(workflow_id: workflow.id, label: 'fieldset_two',
-                   element_type_id: 7, parent_id: top_form.id)
-    Element.create(workflow_id: workflow.id, label: 'input_text2',
-                   element_type_id: 2, parent_id: fs1.id)
+    Element.create(workflow: workflow, label: 'input_text',
+                   element_type: element_types(:text), parent: top_form)
+    fs1 = Element.create(workflow: workflow, label: 'fieldset_one',
+                         element_type: element_types(:fieldset),
+                         parent: top_form)
+    Element.create(workflow: workflow, label: 'fieldset_two',
+                   element_type: element_types(:fieldset), parent: top_form)
+    Element.create(workflow: workflow, label: 'input_text2',
+                   element_type: element_types(:text), parent: fs1)
     a = []
     top_form.traverse { |e, f| a << "#{e.label}#{f}" }
-    assert_equal(5, a.size)
-    assert_equal('form', a[0])
-    assert_equal('input_text', a[1])
-    assert_equal('fieldset_one', a[2])
-    assert_equal('input_text2', a[3])
-    assert_equal('fieldset_two', a[4])
+    assert_equal(15, a.size)
+    assert_equal('form_open', a[0])
+    assert_equal('form', a[1])
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.